[feat] feed page 스타일 추가

migugin opened the pull request:
- 하단 버튼 fixed 작업 필요...

migugin committed:
[feat] PR 피드백 반영

- label 요소에 입력될 값을 받는 props 이름 변경 (label -> labelText)
- 하위 컴포넌트 props의 required, default 프로퍼티 지정
- AppButton.vue: handleClickEvent 이벤트 핸들러 methods로 분리

migugin committed:
[feat] feed page 스타일 추가

MANAPIE commented on index.html:
```diff
@@ -5,6 +5,7 @@
     <link rel="icon" href="/favicon.ico" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>Challenge - Vue2</title>
+    <link rel="stylesheet" href="./src/styles/index.css" />
```
빌드된 후에는 이 주소로 사용 못 해요.
전역 스타일은 여기에 넣는 게 아닙니다.
빌드 옵션에 넣거나 최상위인 App.vue에서 임포트하세요.
*그리고 스타일을 vue 파일에 넣지 말고 scss 파일을 따로 뽑아서 사용해보세요. 다음 달부터는 이렇게 사용합니다. (같은 폴더에 넣는 걸 권장)

migugin replied:
네 빌드 해봤는데 빌드 안되네요
수정하겠습니다!
오늘 SCSS도 사용해보겠습니다~

MANAPIE commented on src/components/AppButton.vue:
```diff
@@ -1,23 +1,22 @@
-    :class="className" 
-    @click="handleClickEvent"
-  >
+  <button :type="type" :class="className" @click="handleClickEvent">
```
프리티어가 그런 거니 이제 이런 건 넘어갈게요

migugin replied:
✅ 😢

MANAPIE commented on src/components/AppButton.vue:
```diff
@@ -27,3 +26,9 @@ export default {
 };
 </script>
+
+<style>
```
해당 컴포넌트에서 쓸 땐 scoped 사용
컴포넌트에서 전역 스타일을 쓰면 관리가 안 됩니다.

migugin replied:
scoped 누락되어서 추가했습니다

MANAPIE commented on src/components/AppInput.vue:
```diff
@@ -33,10 +41,52 @@ export default {
+      type: String,
+      required: true,
+    },
+    inputClass: {
```
저 궁금한 게 있는데 컴포넌트 안에 있는 요소의 클래스를 컴포넌트 밖에서 클래스로 지정해주면 어떻게 쓰이나요?
부모에서 자식의 요소를 다루려는 거면 굳이 클래스를 쓸 필요가 없는데.. 궁금해서요

migugin replied:
말씀하신대로 컴포넌트 안에 있는 요소라서 클래스 필요 없는게 맞습니다
수정하겠습니다

migugin replied:
label 클래스도 빼려고 했는데 a11yhidden 클래스 잡혀있어서 일단 inputClass만 제거하겠습니다

MANAPIE replied:
아 a11yhidden 얜 또 따로 스트링으로 넣어줘야해요?
문서를 꼭 잘 써주시고 읽게 해야겠네요

migugin replied:
다른 방법으로 전달할 수 있게 해야겠습니다

a11yhidden props를 클래스에 String으로 입력하는게 아니라 Boolean 으로 받을 수 있게 수정하려구용

MANAPIE replied:
a11yHidden 이런 컨벤션 맞는 이쁜 이름으로 해주세여~

MANAPIE commented on src/components/AppTextarea.vue:
```diff
@@ -20,15 +20,33 @@ export default {
-    value: String,
+    id: {
+      type: String,
+      required: true,
```
쓸 때마다 아이디를 꼭 넣어줘야 한다구요?
라벨이랑 인풋이랑 매칭만 시킬 용도면 그냥 컴포넌트의 유니크 아이디를 쓰게 한다든가 해서 자동으로 매치해도 되지 않을까요?

migugin replied:
코드 내에 들어가는 상수까지 모듈로 분리했을 때, 가장 직관적으로 input의 역할을 확인할 수 있는게 id라서 명시해주는 게 좋다고 생각하는데 상위 컴포넌트에서 역할이 드러난다면 유니크 아이디를 자동으로 생성해서 연결하는 것도 괜찮을 것 같아요 🤔 🤔

사실 지금 코드에 구현된 props가 마음에 안들어서 props 관리 방법 고려하면서 같이 생각해보겠습니다

MANAPIE replied:
꼼꼼한 개발자면 모르겠는데 아이디를 중복으로 넣고 어리둥절해 하는 것도 종종 봐서요
전역으로 아이디의 중복을 체크해줄 수 없다면 유니크한 아이디만 사용하거나 추가해주는 게 좋을 거 같아요

MANAPIE replied:
다음에 추가하기로

MANAPIE commented on src/components/FeedView/SearchBox.vue:
```diff
@@ -106,3 +105,52 @@ export default {
 };
 </script>
+
+<style>
```
해당 컴포넌트에서 쓸 땐 scoped 사용
컴포넌트에서 전역 스타일을 쓰면 관리가 안 됩니다.

migugin replied:
✅

MANAPIE commented on src/pages/FeedView.vue:
```diff
@@ -1,19 +1,104 @@
+
+    <div class="bottom-sticky-container">
+      <AppButton class="new-post-button" @click="handleNavigateToWrite">
+        Add new post
```
위에는 Open Search 해서 대문자로 시작하는 단어들인데 여기는 아닌 게 신경쓰여요

migugin replied:
불편하네요.. 수정하겠습니다

migugin committed:
[feat] PR 피드백 반영 (6/14)

- style scoped 누락 수정
- AppInput.vue: class props 제거 (추후 label class props 제거도 고려할 것)
- index.css: import 방법 변경
- FeedView.vue: css 작업 완료
- db.json: Mocking Data 추가
